eth_account/account.py

```diff
@@ -169,9 +186,12 @@ def encrypt(private_key, password):
         else:
             key_bytes = HexBytes(private_key)
 
+        if kdf is None:
+            kdf = __class__.default_kdf  # noqa: F821
```

Comment:
I never knew that `__class__` was available like this within a method bodies!

Reply:
Curious (since I'm new-ish to Python and still learning 😁), what made you think it wouldn't be? I used it since it was a `staticmethod` and referencing the class using `__class__` seemed reasonable as we're not calling a method on the `cls`, which is how I've typically seen methods annotated with `@classmethod` used.

Searching around, I find contradictory things though:
>Static methods are great for utility functions, which perform a task in isolation. **They don’t need to (and cannot) access class data.** They should be completely self-contained, and only work with data passed in as arguments.

but also
>The `@staticmethod` decorator is similar to `@classmethod` in that it can be called from an uninstantiated class object, although in this case there is no `cls` parameter passed to its method. 
>
>Since no `self` object is passed either, that means we also don't have access to any instance data, and thus this method can not be called on an instantiated object either.
> 
>These types of methods aren't typically meant to create/instantiate objects, **but they may contain some type of logic pertaining to the class itself**, like a helper or utility method

I think this works because the method is still defined on the class and the `default_kdf` variable is a global / class variable, but now I'm wondering if this is an anti-pattern or not. 🤔

Reply:
Yeah, I missed this. My preferred approach would be to change the method to a `@classmethod` and use `cls.default_kdf` here. It's a little less magical.

Reply:
@stefanmendoza I'd probably have done `type(self)` but I realize now that this was a `staticmethod`.  @carver suggestion to change this to a class method seems like the right approach.

Reply:
I can put out a quick follow-up PR 👍